module/documents/token.mjs

```diff
@@ -107,6 +107,37 @@ export default class TokenDocument5e extends SystemFlagsMixin(TokenDocument) {
     this.texture.scaleY = this._source.texture.scaleY * dts;
   }
 
+  /* -------------------------------------------- */
+  /*  Movement                                    */
+  /* -------------------------------------------- */
+
+  /**
+   * Set up the system's movement action customization.
+   */
+  static registerMovementActions() {
+    for ( const type of Object.keys(CONFIG.DND5E.movementTypes) ) {
+      const actionConfig = CONFIG.Token.movement.actions[type];
+      if ( !actionConfig ) continue;
+      actionConfig.getCostFunction = (...args) => this.getMovementActionCostFunction(type, ...args);
```

Comment:
Only half speed if the actor doesn't have this movement type.

```suggestion
      actionConfig.getCostFunction = (...args) => this.getMovementActionCostFunction(type, ...args);
      actionConfig.getAnimationOptions = token => {
        const actorMovement = token.actor?.system.attributes?.movement ?? {};
        if ( !(type in actorMovement) || actorMovement[type] ) return {};
        return {movementSpeed: CONFIG.Token.movement.defaultSpeed / 2};
      };
```

Technically the half speed is bit wrong with difficult terrain (slower than it should be). But probably not super important.